pkg‑config file incorrectly adds -ldoctest to Libs

### Description

`doctest` is a header‑only library, so there is no actual library to link against. However, the `doctest.pc.in` file introduced in PR #865 contains:

```pkgconfig
Libs: -L${libdir} -ldoctest
```

This causes any build system (like Meson) that uses `pkg‑config` to detect `doctest` to add `-ldoctest` to the linker command line, which then fails because no such library exists.

### wayfire build

```
Build targets in project: 61

wayfire 0.11.0

  Subprojects
    wf-json           : YES
    wf-touch          : YES
    wf-utils          : YES

  User defined options
    auto_features     : enabled
    b_pie             : true
    buildtype         : plain
    libexecdir        : lib
    prefix            : /usr
    python.bytecompile: 1
    sbindir           : bin
    tests             : disabled
    wrap_mode         : nodownload

Found ninja-1.13.2 at /usr/bin/ninja
INFO: autodetecting backend as ninja                                                                                                                               
INFO: calculating backend command to run: /usr/bin/ninja -C /home/ssfdust/Program/clones/wayfire-pkgbuild/wayfire-git/src/build
ninja: Entering directory `/home/ssfdust/Program/clones/wayfire-pkgbuild/wayfire-git/src/build'
[500/554] Linking target subprojects/wf-touch/test/basic_test
FAILED: [code=1] subprojects/wf-touch/test/basic_test 
c++  -o subprojects/wf-touch/test/basic_test subprojects/wf-touch/test/basic_test.p/basic_test.cpp.o -Wl,--as-needed -Wl,--no-undefined -pie -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -g -fvar-tracking-assignments -ffile-prefix-map=/home/ssfdust/Program/clones/wayfire-pkgbuild/wayfire-git/src=/usr/src/debug/wayfire-git -Wl,--start-group subprojects/wf-touch/libwftouch.a /usr/lib/libglm.a -ldoctest -Wl,--end-group
/usr/bin/ld: cannot find -ldoctest: No such file or directory
collect2: error: ld returned 1 exit status
[505/554] Compiling C++ object plugins/ipc-rules/libipc-rules.so.p/ipc-rules.cpp.o
ninja: build stopped: subcommand failed.
```

### Steps to reproduce

#### meson.build

```
project('repro', 'cpp')

# Use pkg‑config to find doctest (this will read the generated .pc file)
doctest_dep = dependency('doctest', method: 'pkg-config')

# Build a test executable that includes doctest.h
executable('test_app', 'test_app.cpp', dependencies: doctest_dep)
```

#### test_app.cpp

```
#include <doctest/doctest.h>

int main() {
    return 0;
}
```

#### build log

meson setup build

```
meson setup build
The Meson build system
Version: 1.10.2
Source dir: /home/ssfdust/Program/sources/aur/doctest/repro
Build dir: /home/ssfdust/Program/sources/aur/doctest/repro/build
Build type: native build
Project name: repro
Project version: undefined
C++ compiler for the host machine: c++ (gcc 15.2.1 "c++ (GCC) 15.2.1 20260209")
C++ linker for the host machine: c++ ld.bfd 2.46
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: YES (/usr/bin/pkg-config) 2.5.1
Run-time dependency doctest found: YES 2.5.0
Build targets in project: 1

Found ninja-1.13.2 at /usr/bin/ninja
```

ninja -C build/

```
ninja: Entering directory `build/'
[2/2] Linking target test_app
FAILED: [code=1] test_app 
c++  -o test_app test_app.p/test_app.cpp.o -Wl,--as-needed -Wl,--no-undefined -ldoctest
/usr/bin/ld: cannot find -ldoctest: No such file or directory
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### doctest version

2.5.0

### Operating system

Archlinux

### Architecture

x86_64

### Compiler

c++ (GCC) 15.2.1 20260209

### Context

_No response_

--- a/doctest.pc.in
+++ b/doctest.pc.in
@@ -6,5 +6,4 @@ includedir="${prefix}/include"
 Name: @PROJECT_NAME@
 Description: The fastest feature-rich C++11/14/17/20/23 single-header testing framework
 Version: @PROJECT_VERSION@
-Libs: -L${libdir} -ldoctest
 Cflags: -I${includedir}

